Source/MayaLiveLinkPlugin.cpp

```diff
@@ -272,6 +272,188 @@ struct FStreamHierarchy
 	{}
 };
 
+/*
+*	Utility class for keeping track of user defined attributes from the
+*	channel box in Maya.
+*/
+class FStreamedUserDefinedAttributes
+{
+protected:
+	/*
+	*	This struct is used to cache each attribute MPlug for the streamed 
+	* 	user defined attributes. The struct contains member variables to
+	*	allow dirty-checking to help trigger a refresh of static data
+	*	whenever an attribute property is modified in the channel box 
+	*	in Maya.
+	*/
+	struct FAttributeInfo
+	{
+		// An MPlug is a connection which supports reading values and properties from object attributes
+		MPlug Plug;
+		inline float Value() { return Plug.asFloat(); }
+
+		// The cached partialName of the attribute
+		MString PartialName = "";
+
+		// Determines if an attribute should be streamed via Live Link (see IsPlugRelevantForStream)
+		bool bRelevantForStream = false;
+
+		// Determines if an attribute property has been modified by the user
+		bool HasDirtyProperties() const
+		{
+			return (PartialName != Plug.partialName()) || (bRelevantForStream != IsPlugRelevantForStream(Plug));
+		}
+	};
+
+	// The total number of attributes available on the root joint of the streamed hierarchy
+	int NumAttributesOnRootJoint = 0;
+
+	// The sub-set of attributes added by the user to the root joint
+	TArray<FAttributeInfo> UserDefinedAttributes;
+
+public:
+	/*
+	*	Determines if a user defined attribute should be streamed via Live Link.
+	*	Note that MPlug is used for querying properties and not MFnAttribute. 
+	*	Even though both has the same functions defined, only MPlug replicates the 
+	*	behavior of getAttr in MEL. MFnAttribute behaves as attributeQuery in MEL,
+	*	which do not report the state of the attribute that is needed.
+	*/
+	static bool IsPlugRelevantForStream(MPlug Plug)
```

Comment:
We should verify the type of the plug here. Only using keyable is good. We do somewhat the samething in Mobu where we only use animatable properties but also we check if the property is of a good type (float, enum, int, etc...). Properties in livelink data are always float so your Plug.AsFloat is fine but it won't work for String data type for example. Adding a filter for type would fix this.